cmd/gopass-jsonapi/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+
+	"github.com/gopasspw/gopass/pkg/gopass"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	name = "gopass-jsonapi"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// trap Ctrl+C and call cancel on the context
+	ctx, cancel := context.WithCancel(ctx)
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt)
+	defer func() {
+		signal.Stop(sigChan)
+		cancel()
+	}()
+	go func() {
+		select {
+		case <-sigChan:
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
+
+	gp, err := gopass.New(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	ja := &jsonapiCLI{
+		gp: gp,
+	}
+
+	app := cli.NewApp()
+	app.Name = name
+	app.Version = "TODO"
+	app.Usage = "Setup and run gopass-jsonapi as native messaging hosts, e.g. for browser plugins"
+	app.EnableBashCompletion = true
+	app.Action = func(c *cli.Context) error {
+		if strings.HasSuffix(os.Args[0], "native_host") || strings.HasSuffix(os.Args[0], "native_host.exe") {
+			return ja.listen(c)
+		}
+		return cli.ShowAppHelp(c)
+	}
+	app.Commands = []*cli.Command{
+		{
+			Name:        "listen",
```

Comment:
If we have the ability to split up this into two binaries, I would propose to have a separate binary for the api listener. That would also make the windows naming hack obsolete (which has been removed in this PR already) and prevent accidental calling of jsonapi listen.

Reply:
I don't mind splitting up the listener further.
No need to do this as part of this PR, but once we merge this you can go ahead and do that.

Reply:
Sure, let's keep it as it is - whatever is simpler.